internal/observability/call.go

```diff
@@ -227,6 +227,9 @@ func (c call) endLogs(
 	fields = append(fields, zap.Duration("latency", elapsed))
 	fields = append(fields, zap.Bool("successful", err == nil && !isApplicationError))
 	fields = append(fields, c.extract(c.ctx))
+	if deadlineTime, ok := c.ctx.Deadline(); ok {
```

Comment:
Sorry for a drive-by-review. A question:

The PR description says
> Log original context deadline

But here you're logging not a deadline (which is a timestamp), but a duration. 

Generally, deadline seems to be tied to exact timestamps, like https://pkg.go.dev/context#WithDeadline. 

This seems even stronger in the RPC world where per https://grpc.io/blog/deadlines/:
> Some language APIs work in terms of a **deadline**, a fixed point in time by which the RPC should complete. Others use a **timeout**, a duration of time after which the RPC times out.

If I read this PR correctly, you're logging `timeout` but calling it `deadline`. Which one did you mean? Is it perhaps not too late to rename the field before we bump the version?

cc @jronak

Reply:
Oh, I found https://github.com/yarpc/yarpc-go/pull/2189 now. This is great, sorry for the noise :)